modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2925,8 +2951,14 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 						break;
 				}
 			}
-			outer = outer->outer;
+			if (script_class->outer != nullptr) {
+				script_classes.insert(script_class->outer);
```

Comment:
i don't know how to request changes, but i'm pretty sure this should be a loop inside the next if statement, collecting only this outer class's bases outer classes (the terminology is so confusing), same in `resolve_datatype`